### DISCUSSION TITLE
Datagrid Double Validation?

### DISCUSSION THREAD

**Discussion by SamCharette** [Megabit/Blazorise#1594] [Q&A]
Hi there!  I just started using Blazorise (awesome), so I'm still missing a few concepts I'm sure.  I made a simple datagrid that uses the popup edit method.  I wanted to do some custom validation, so I added Validator="@CheckName" to the field, and a function that checks for valid results.

It works, but when I click save it highlights the invalid field and gives a generic "One or more fields have an error. Please check and try again." error.  If I click save again then the custom validation text appears.

Is there a way to go straight to the custom validation errors and skip the generic one?

**Comment by stsrki** [comment]
Maybe you just need to set `ErrorText`

Try:

```cs
void YourMethod(ValidatorEventArgs eventArgs)
{
  eventArgs.ErrorText = "Your error message";

  // other code
}
```

It's hard to tell without any code sample. If that doesn't help, can you create a small sample to reproduce the issue?

**Reply by SamCharette** [reply]
Unfortunately error text does get set.  Here's the validation function (I'm new to posting on github, so may not get this quite right)

```c#
private void CheckName(ValidatorEventArgs validationArgs)
        {

            ValidationRule.IsNotEmpty(validationArgs);

            if (validationArgs.Status == ValidationStatus.Error)
            {
                validationArgs.ErrorText = "Name is required.";
            }
            else
            {
                string name = (string)validationArgs.Value;
                if (Db.Statuses.Any(x => x.Name == name))
                {
                    validationArgs.Status = ValidationStatus.Error;
                    validationArgs.ErrorText = "Name must be unique.";
                }
            }
        }
```

The proper validation does appear, but only hitting save a second time.  The example datagrid does the same:

https://bootstrapdemo.blazorise.com/tests/datagrid

**Comment by stsrki** [comment]
It seems this is a bug indeed. Can you open a new issue and I will try to fix it for the 0.9.3 preview.

**Reply by SamCharette** [reply]
Sure thing :)
